File: 2024/AdventOfCode/Tests/Day06Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day06Test {
    private const string Example = "Inputs/06-Example.txt";
    private const string Input = "Inputs/06.txt";
    private static Day06 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day06(Example);
            const string expected = "41";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day06(Input);
            const string expected = "4778";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample() {
            _day = new Day06(Example);
            const string expected = "6";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day06(Input);
            const string expected = "1618";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}

File: 2025/test/advent_of_code/day_06_test.exs
defmodule AdventOfCode.Day06Test do
  use ExUnit.Case

  import AdventOfCode.Day06

  @day 6
  @year 2025

  @example1 String.trim("""
            123 328  51 64
             45 64  387 23
              6 98  215 314
            *   +   *   +
            """)

  test "part1 example1" do
    result = part1(@example1)

    assert result == 4277556
  end

  test "part1" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part1(input)

    assert result == 5361735137219
  end

  test "part2 example1" do
    result = part2(@example1)

    assert result == 3263827
  end

  test "part2" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part2(input)

    assert result == 11744693538946
  end
end
